### PR TITLE
improve filter input

### LINKED ISSUE #55: Improved filter input
Currently, we have a fixed-size input field limiting usability when providing multiple filter rules.
Look at [Greynoise](https://viz.greynoise.io/ip/8.8.8.8) input field for inspiration.

- On focus, expand the input field's width
- Add a clear or reset button to the expanded input field to remove all filters
- Add additional lines to the input field if the text exceeds the field length

### PR DESCRIPTION
**Pull request by mushroomgenie**
Fixes #55

### CODE REVIEW

**Comment by mushroomgenie** [comment]
@glaslos Was able to increase the width and add a reset query button on focus. Needed some clarity on the multiline input.

**Comment by glaslos** [comment]
Nice work! Regarding your question: Currently if you exceed the width of your input field, the text is pushed to the left out of the input field. I'd suggest wrapping the text and adding a new line. 
We should also "shrink" the field once it loses focus to not impact visibility on the page.

File: src/components/Filter.svelte [hunk 1]
```diff
@@ -6,6 +6,11 @@
 
     let filter: string = '';
     let filterValid: boolean = false;
+    let focus: boolean = false;
+
+    function handleFocus() {
```
Comment (glaslos) [inline src/components/Filter.svelte:11]: For very simple handlers you can also define them [inline](https://svelte.dev/tutorial/inline-handlers)

**Comment by dkumiszhan** [comment]
I am not sure if making input expandable now is useful. In the original version and [Greynoise](https://viz.greynoise.io/ip/8.8.8.8) the query is within the header and by default does not occupy the whole line. There, it might be useful to expand the input. In the current version, there is nothing else on the same line next to the input. So I think we can consider 2 options:

1. Move the filter input back to the header and do the expandable approach similar to what you are doing.
2. Keep the filter input where it is but make it more expanded by default without the dynamic expanding. 

Also it doesn't seem to apply filter when the input is expanded and "Reset Query" button  doesn't remove all filters. I like how the reset button shows up on [Greynoise](https://viz.greynoise.io/ip/8.8.8.8). It would be nice if we could do something like that.
 
Could you also please fix the commits history, and rebase your code on top of what is already merged?

**Comment by glaslos** [comment]
@mushroomgenie how about going for option 2 suggested by @dkumiszhan ?

**Comment by mushroomgenie** [comment]
@glaslos sure we can do that.

**Comment by glaslos** [comment]
Closing due to inactivity

**Pull request closed without merging**